src/by/epam/ayem/module5/service/TreasureHouseService.java

```diff
@@ -0,0 +1,125 @@
+package by.epam.ayem.module5.service;
+
+/*Задача 4.
+Создать консольное приложение, удовлетворяющее следующим требованиям:
+1) Приложение должно быть объектно-, а не процедурно-ориентированным.
+2) Каждый класс должен иметь отражающее смысл название и информативный состав.
+3) Наследование должно применяться только тогда, когда это имеет смысл.
+4) При кодировании должны быть использованы соглашения об оформления кода java code convention.
+5) Классы должны быть грамотно разложены по пакетам.
+7) Консольное меню должно быть минимальным.
+8) Для хранения данных можно использовать файлы.
+
+Дракон и его сокровища. Создать программу, позволяющую обрабатывать сведения о 100 сокровищах в пещере дракона.
+Реализовать возможность просмотра сокровищ, выбора самого дорогого по смоимости сокровища и выбора сокровищ
+на заданную сумму.*/
+
+import by.epam.ayem.module5.model.*;
+
+import java.util.List;
+import java.util.Random;
+import java.util.Scanner;
+
+public class TreasureHouseService {
+
+    private Scanner scanner = new Scanner(System.in);
+    private Reader reader = new Reader();
+    private Writer writer = new Writer();
+    private Random random = new Random();
+    private final static String FILE_NAME = "treasures.txt";
+
+    public void findTreasuresAmountingTo(TreasureHouse treasureHouse) {
+        System.out.println("Please, enter the worth of the treasures: ");
+
+        while (!scanner.hasNextDouble()) {
+            scanner.next();
+            System.out.println("Please, use a number.");
+        }
+
+        double worth = scanner.nextDouble();
+        double totalAmount = 0.0;
+        double left = worth;
+        System.out.println("Treasures amounting to " + worth + ":");
```

Comment:
If id debug information that please use logger

Reply:
Sorry, didn't get it.

Reply:
logger.info() is better than System.out.println